src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Value/DisallowedTypeConverterFactory.cs

```diff
@@ -23,7 +23,13 @@ public override bool CanConvert(Type type)
                 // Invoking such ctors is not safe when used with untrusted user input.
                 type == typeof(SerializationInfo) ||
                 type == typeof(IntPtr) ||
-                type == typeof(UIntPtr);
+                type == typeof(UIntPtr) ||
+                // To be added in future releases; guard against invalid object-based serializations.
+                // https://github.com/dotnet/runtime/issues/53539
+                IsDateOnlyOrTimeOnly(type);
+
+            static bool IsDateOnlyOrTimeOnly(Type type)
+                => type.Assembly == typeof(int).Assembly && type.FullName is "System.DateOnly" or "System.TimeOnly";
```

Comment:
Why can't this use `typeof(DateOnly)`/`typeof(TimeOnly)`?

Reply:
We also target netstandard2.0

Reply:
What's the end of support date for netstandard2.0? isn't it very soon? https://dotnet.microsoft.com/platform/support/policy/dotnet-core - if that's a case it might be worth considering dropping the support

Reply:
> We also target netstandard2.0. I thought using one stringly typed approach for both net6.0 and netstandard2.0 would make things simpler from a testing perspective.

I'd rather do the most efficient thing we can do for the given platform.

Reply:
What would we be testing if the types won't be used in that build anyway?

Reply:
Testing for the pathological scenario where a >= net6.0 runtime references a netstandard2.0 build of STJ. Since getting that in place is expensive infrastructure wise, I'm proposing we do it by proxy in net6.0 Debug builds.

Reply:
Alternatively the converter factory could look up and cache the Type instances to minimize the amount of work needed per converter construction:
```csharp
private readonly Type? _dateOnlyType = typeof(int).Assembly.GetType("System.DateOnly");
private readonly Type? _timeOnlyType = typeof(int).Assembly.GetType("System.TimeOnly");

...
public bool CanConvert(Type typeToConvert) => ... || typeToConvert == _dateOnlyType || typeToConvert == _timeOnlyType;
```

Reply:
No :-) That would make the implementation more expensive for everyone.  I don't think we should be bending over backwards to find ways to make the netstandard2.0 build fail when exposed to types not available via netstandard2.0. I'm OK with the proposed idea if you really think it's critical,  but I'd just assume leave it out.

Reply:
(Note that someone using the existing netstandard2.0 asset can and will continue doing so unless they explicitly upgrade, so they're already not getting this exception.)

Reply:
I'd probably be less concerned if doing nothing for netstandard2.0 resulted in `DateOnly` serialization failing in some weird way. But in any case this is addressing a theoretical corner case so erring on the side of simplicity seems like a reasonable approach.

cc @layomia @ericstj for second opinions.

Reply:
> the pathological scenario where a >= net6.0 runtime references a netstandard2.0 build of STJ >= 6.0.0.

That is incorrectly constructed app. You would have to bend backwards to the make SDK to produce app like that; or manually copy files around. 

I agree with @stephentoub that this is not something we need to worry about. It is unsupported and untested scenario.

Reply:
I'm assuming there's also a trimming angle to be considered here.